Machine ID: TBot v14 documentation changes (#27460)
gravitational/teleport: docs/pages/machine-id/reference/v14-upgrade-guide.mdx

---
title: Machine ID V14 Upgrade Guide
description: Upgrade instructions for Machine ID in Teleport 14.0
---

Teleport 14.0 introduces a new version of the configuration format (v2) for
the Machine ID agent, `tbot`. This guide explains the breaking changes present
between the v1 and v2 configuration format and how to migrate.

> You do not need to take any action if:

Author: I'd be surprised if that was possible - the `destinations`/`outputs` change will affect basically any Machine ID config file.

- You do not use Machine ID.
- You configure `tbot` using only the CLI parameters.

> ## Overview

Author: d4bc33f05b

---
title: Machine ID V14 Upgrade Guide
description: Upgrade instructions for Machine ID in Teleport 14.0
---

Teleport 14.0 introduces a new version of the configuration format (v2) for
Machine ID's agent `tbot`. This guide explains the breaking changes present
between the v1 and v2 configuration format and how to migrate.

You do not need to take any action if:

- You do not use Machine ID.
- You configure `tbot` using only the CLI parameters.

## Overview

Previously, the `tbot` configuration contained a list of destinations. A
`destination` described what should be written, where it should be written and
what credential attributes should be used.

> Each `destination` had a large number of configuration fields, and many of

Author: I really like this suggested wording ! Thank you!

  roles:
  - editor
  app: grafana
```

**Example of new configuration**
```yaml
version: v2
outputs:
- type: application
  roles:
  - editor
  destination:
    type: directory
    path: /opt/machine-id
```

See the [configuration reference](./configuration.mdx) for details of the new
configuration version and outputs.

## Migrating configuration files

On start, `tbot` will automatically attempt to migrate a v1 configuration to
v2 in order to continue without disruption. This migration is ephemeral and
is not written to disk.

We heavily recommend converting your configuration file from v1 to v2
permanently. A warning will be emitted when starting `tbot` until you do so.

Determine the path of your current configuration file and the path you wish to
write the migrated configuration file to. Then run:

```
tbot migrate -c /config.yaml -o /config.migrated.yaml
```

Inspect the migrated configuration file to ensure it meets your expectations,
and try running `tbot` with it. If everything is ok, you may replace your
original configuration file with the migrated one.

### Migration failure

Consider the error logged by the migration command - it likely points out which

> part of your original configuration cannot be migrated. It may be that your

Reviewer: ```suggestion
your original configuration can't be migrated. It might be that your
```
